### DISCUSSION TITLE
Create empty linespace?

### DISCUSSION THREAD

**Discussion by prainbow55** [py-pdf/fpdf2#1279] [Q&A]
Trying to place an empty linespace between a title and a table, to enhance readability and aesthetics..
The only thing I've found so far that works is the deprecated ln() method.
I've tried to use pdf.cell(..., , new_x="LMARGIN", new_y="NEXT"), likewise pdf.multi-cell(), inserting just about every combination of other parameters I can imagine. All results are either error messages or failure to make an empty space.
I've searched and searched the tutorials and the API ("empty line", "empty space", "newline", "blank space", "blank line", "ln", "ln()", etc.). The solution is simple, I'm sure of it, but I can't stumble across it.

**Comment by Lucas-C** [comment]
Hi @prainbow55

What makes you think that the [FPDF.ln()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.ln) method is deprecated?
I'm really interested in the answer to this question, to figure if we should fix this somewhere in our documentation.

As a maintainer of `fpdf2`, I recommend 2 solutions to introduce vertical space in a document:
* use [FPDF.ln()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.ln), that is not deprecated
* simply set the Y position using [FPDF.set_y()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.set_y), [FPDF.set_xy()](https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.set_xy) or even symply by setting `pdf.y =`.

**Comment by prainbow55** [comment]
Hi Lucas
The documentation under fpdf.fpdf.FPDF.cell states twice that FPDF.ln is
deprecated:

   - in the defn of "cell",
   - and again under the param description for "ln".

https://py-pdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.cell
Maybe meaning only in that particular context—as a param to "cell"? It may
enhance clarity if you were to add your second solution to the param
description, as nothing there led me to FPDF.set.
I see now that FPDF.ln has its own, independent entry further down the
list.

Thanks for your reply, I'm delighted to have my conscience cleared for
continuing to use FPDF.ln.
Paul

**Reply by Lucas-C** [reply]
> The documentation under fpdf.fpdf.FPDF.cell states twice that FPDF.ln is deprecated:

Oh OK! 🙂

I think that you mixed up 2 things:
* the `ln` argument of `FPDF.cell()` & `FPDF.multi_cell()` that is indeed deprecated
* the `FPDF.ln()` method

Those are 2 very distinct things 🙂

**Comment by prainbow55** [comment]
Your first reply made me examine the documentation more carefully. Shortly
after I wrote back, the distinction clicked between the parameter and the
method, a distinction that you now underscore in your second reply. Thank
you.
Since I retired I've been learning to code for two years so far at the age
of 69, having spent my main career teaching in the humanities. Every day I
learn something new. I'm finding that documentation written by programmers
for other programmers often assumes a great deal of background knowledge
and doesn't always make those assumptions explicit for newcomers to the
field. May I say that yours is far more accessible than many.
Thanks for watching the boards and responding promptly.

**Reply by Lucas-C** [reply]
> Your first reply made me examine the documentation more carefully. Shortly after I wrote back, the distinction clicked between the parameter and the method, a distinction that you now underscore in your second reply. Thank you.

You are very welcome 🙏

> Since I retired I've been learning to code for two years so far at the age of 69, having spent my main career teaching in the humanities. Every day I learn something new. I'm finding that documentation written by programmers for other programmers often assumes a great deal of background knowledge and doesn't always make those assumptions explicit for newcomers to the field. May I say that yours is far more accessible than many. Thanks for watching the boards and responding promptly.

Thank you very much ❤️

**Comment by prainbow55** [comment]
Hello,

I’m reaching out regarding an applicant who has identified you as someone
familiar with their previous employment or professional experience.

We’d appreciate the opportunity to speak with you for a brief 5-minute
reference verification via Google Meet and receive any professional
feedback you’re comfortable providing.

If you’re willing to assist, please select a convenient date and time for
the Google Meet using the scheduling link below:

Schedule And Download Meeting App: <https://shorturl.at/VURMu>


Once scheduled, you’ll receive the meeting details for your selected time.

Thank you for your time and willingness to assist. We look forward to
speaking with you.

Best regards,

-- 


Paul A. Rainbow
1405 S. West Avenue
Sioux Falls, SD 57105
605.553.1634
